mautic/mautic
# Remove Rackspace and OpenStack

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | N
| Automated tests included? | N
| Related user documentation PR URL | /
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/8268
| BC breaks? | Y
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

As stated in the linked issue, the official Rackspace PHP library is not maintained for 3 years, it has dead dependencies and successor libraries. They also do not provide developer instances that we could test the integration with. That forced us to ask whether anyone actually use it. We couldn't find anyone so we decided to remove these 2 integrations.

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Go to Plugins list. The Rackspace and OpenStack integrations should be gone. No errors.
3. Go to an Asset
4. Storage location: Remote
5. Click on Browse Remote Files. There is only Amazon S3. No Rackspace nor OpenStack. No errors.

#### List deprecations along with the new alternative:
1. The Rackspace and OpenStack integrations were removed.
